### PR TITLE
Adding Pascal language references to books

### PR DESCRIPTION
**Pull request by qxzsilver1**
## What does this PR do?
Adding Pascal

### Description
Adding Pascal language references to books

### Why is this valuable (or not)?
There are currently no Pascal references for programming languages.

### How do we know it's really free?
All the links are free and accessible.

### For book lists, is it a book? For course lists, is it a course? etc.
This is a list of books/references.

## Checklist:
- [x] Read our [contributing guidelines](https://github.com/EbookFoundation/free-programming-books/blob/main/docs/CONTRIBUTING.md)
- [x] [Search](https://ebookfoundation.github.io/free-programming-books-search/) for duplicates.
- [x] Include author(s) and platform where appropriate.
- [x] Put lists in alphabetical order, correct spacing.
- [x] Add needed indications (PDF, access notes, under construction)

## Follow-up

- Check the status of GitHub Actions and resolve any reported warnings!

### CODE REVIEW

File: books/free-programming-books-langs.md [hunk 1]
```diff
@@ -1653,6 +1654,19 @@ That section got so big, we decided to split it into its own file, the [BY SUBJE
 * [TrueOS® Users Handbook](https://www.trueos.org/handbook/trueos.html)
 
 
+### Pascal
+
+* [Free Pascal Reference guide](https://www.freepascal.org/docs-html/ref/ref.html)
+* [Modern Object Pascal Introduction for Programmers](https://castle-engine.io/modern_pascal_introduction.html) (HTML)
+* [Pascal Language Reference](https://docs.oracle.com/cd/E19957-01/802-5762/802-5762.pdf) (PDF)
+* [Pascal Programming Reference Manual](https://public.support.unisys.com/aseries/docs/clearpath-mcp-17.0/pdf/86000080-103.pdf) (PDF)
```
Comment (eshellman) [inline books/free-programming-books-langs.md:1662]:
```suggestion
* [Pascal Programming Reference Manual](https://public.support.unisys.com/aseries/docs/clearpath-mcp-17.0/pdf/86000080-103.pdf) - Unisys (PDF)
```

File: books/free-programming-books-langs.md [hunk 2]
```diff
@@ -1653,6 +1654,19 @@ That section got so big, we decided to split it into its own file, the [BY SUBJE
...
+* [Pascal Quick Reference](https://ksvi.mff.cuni.cz/~dingle/2017/pascal_reference.html)
+* [Turbo Pascal Reference Guide](http://bitsavers.org/pdf/borland/turbo_pascal/Turbo_Pascal_Version_5.0_Reference_Guide_1989.pdf) (PDF)
```
Comment (eshellman) [inline books/free-programming-books-langs.md:1664]:
```suggestion
* [Turbo Pascal Reference Guide (1989)](http://bitsavers.org/pdf/borland/turbo_pascal/Turbo_Pascal_Version_5.0_Reference_Guide_1989.pdf) - Borland International (PDF)
```

File: books/free-programming-books-langs.md [hunk 3]
```diff
@@ -1653,6 +1654,19 @@ That section got so big, we decided to split it into its own file, the [BY SUBJE
...
+* [Vector Pascal, an Array Language](http://www.dcs.gla.ac.uk/~wpc/reports/compilers/compilerindex/vp-ver2.html)
```
Comment (eshellman) [inline books/free-programming-books-langs.md:1665]:
```suggestion
* [Vector Pascal, an Array Language](http://www.dcs.gla.ac.uk/~wpc/reports/compilers/compilerindex/vp-ver2.html) - Paul Cockshott, Greg Michaelson
```

File: books/free-programming-books-langs.md [hunk 4]
```diff
@@ -1653,6 +1654,19 @@ That section got so big, we decided to split it into its own file, the [BY SUBJE
...
+* [Vector Pascal Reference Manual](https://www.researchgate.net/publication/220177664_Vector_Pascal_reference_manual) (PDF)
```
Comment (eshellman) [inline books/free-programming-books-langs.md:1666]: In general, we do not link to file sharing sites, or file hosting sites such as google docs. or dropbox. (or researchgate)

File: books/free-programming-books-langs.md [hunk 5]
```diff
@@ -1653,6 +1654,19 @@ That section got so big, we decided to split it into its own file, the [BY SUBJE
...
+* [VSI Pascal for OpenVMS Reference Manual](https://vmssoftware.com/docs/VSI_PASCAL_REF.pdf) (PDF)
```
Comment (eshellman) [inline books/free-programming-books-langs.md:1667]:
```suggestion
* [VSI Pascal for OpenVMS Reference Manual](https://vmssoftware.com/docs/VSI_PASCAL_REF.pdf) - VMS Software (PDF)
```

**Comment by eshellman** [comment]
long over due addition!

File: books/free-programming-books-langs.md (hunk 4 above)
Comment (qxzsilver1) [reply books/free-programming-books-langs.md:1666]: I can get rid of these sources - I thought they were good references, but I understand that file sharing/hosting is generally unreliable as long-term references.

**Comment by eshellman** [comment]
Thanks, and welcome to free-programming-books!

**Pull request merged**